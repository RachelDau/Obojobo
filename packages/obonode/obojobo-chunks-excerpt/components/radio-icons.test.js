import RadioIcons from './radio-icons'
import renderer from 'react-test-renderer'
import React from 'react'
import { mount } from 'enzyme'

describe('Radio Icons', () => {
	const options = [
		{ label: 'small', icon: 'image' },
		{ label: 'medium', icon: 'image' },
		{ label: 'large', icon: 'image' }
	]

	const selectedOption = 'medium'

	const ariaLabel = 'sizes'

	const onChange = jest.fn()

	afterEach(() => {
		jest.clearAllMocks()
	})

	test('Node builds the expected component', () => {
		const component = renderer.create(
			<RadioIcons
				name="sizes"
				ariaLabel={ariaLabel}
				options={options}
				selectedOption={selectedOption}
				onChangeOption={onChange}
			/>
		)

		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('Handles mouse down event', () => {
		const component = mount(
			<RadioIcons
				name="sizes"
				ariaLabel={ariaLabel}
				options={options}
				selectedOption={selectedOption}
				onChangeOption={onChange}
			/>
		)

		// console.log(component.debug());

		component
			.find('label')
			.at(0)
			.simulate('mousedown')

		expect(onChange).toHaveBeenCalled()
		expect(onChange).toHaveBeenCalledWith('small')
	})

	test('Handles change event', () => {
		const component = mount(
			<RadioIcons
				name="sizes"
				ariaLabel={ariaLabel}
				options={options}
				selectedOption={selectedOption}
				onChangeOption={onChange}
			/>
		)

		component
			.find('input')
			.at(0)
			.simulate('change')

		expect(onChange).toHaveBeenCalled()
		expect(onChange).toHaveBeenCalledWith('small')
	})
})
