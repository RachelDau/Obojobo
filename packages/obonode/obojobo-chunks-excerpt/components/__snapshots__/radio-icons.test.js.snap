// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Icons Node builds the expected component 1`] = `
<div
  aria-label="sizes"
  className="radio-icons"
  role="radiogroup"
>
  <div
    className="options"
  >
    <label
      className="is-not-selected is-option-small"
      onMouseDown={[Function]}
      style={
        Object {
          "transform": "translate(0px, 0)",
        }
      }
    >
      <input
        checked={false}
        name="sizes"
        onChange={[Function]}
        type="radio"
        value="small"
      />
      <span>
        small
      </span>
      image
    </label>
    <label
      className="is-selected is-option-medium"
      onMouseDown={[Function]}
      style={
        Object {
          "transform": "translate(-1px, 0)",
        }
      }
    >
      <input
        checked={true}
        name="sizes"
        onChange={[Function]}
        type="radio"
        value="medium"
      />
      <span>
        medium
      </span>
      image
    </label>
    <label
      className="is-not-selected is-option-large"
      onMouseDown={[Function]}
      style={
        Object {
          "transform": "translate(-2px, 0)",
        }
      }
    >
      <input
        checked={false}
        name="sizes"
        onChange={[Function]}
        type="radio"
        value="large"
      />
      <span>
        large
      </span>
      image
    </label>
  </div>
</div>
`;
